pkg/proxy/proxy_test.go

```diff
@@ -92,6 +99,24 @@ func TestListNamespacedRelease(t *testing.T) {
 	}
 }
 
+func TestListOldRelease(t *testing.T) {
+	app := AppOverview{"foo", "1.0.0", "my_ns", "icon.png", "DEPLOYED"}
+	appUpgraded := AppOverview{"foo", "1.0.1", "my_ns", "icon.png", "DEPLOYED"}
+	proxy := newFakeProxy([]AppOverview{app, appUpgraded})
+
+	// Should avoid old release versions
+	releases, err := proxy.ListReleases(app.Namespace, 256)
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+	if len(releases) != 1 {
+		t.Errorf("It should return a sivnle release")
+	}
+	if !reflect.DeepEqual([]AppOverview{appUpgraded}, releases) {
+		t.Errorf("Unexpected list of releases %v", releases)
+	}
+}
+
```

Comment:
Please also add a test case where we have a mix of DEPLOYED and FAILED states in the history of a release and still only return the latest version

Reply:
The problem with that is that the `fake` helm client doesn't behave like the real one. It just returns everything you add (without filtering SUPERSEDED for example). The bug is covered but the previous test `TestListOldRelease` in which we test that when Tiller returns several releases with the same ID the proxy will just return the latest but I can change the status names to reflect clearer the bug we are fixing.

Reply:
@andresmgot ah I see, yeah that sucks. I think it doesn't really matter then, thanks!